Tutorial ts

Vaishnavishah opened the pull request:
Added content to typescript basics tutorial. Removed TODOs.

jon-bell commented:
Can you also add some discussion of Array functions, in particular, `map`, `filter`, and `forEach`? Our linter rules strongly discourage most for loops that could be replaced by calls to these functions, and it would be useful for those alternatives to be included in the tutorials.

You can continue to push commits to your same branch `tutorial-ts`, and they will appear in this pull request. The deployment preview will also continue to update.

Thanks for your contribution!

Vaishnavishah commented:
Fixed the links, removed TODOs, fixed any typos and formatting issues, updated the content according to the suggestions, added Map, Reduce, Filter, ForEach.
